src/Framework/LogInterpolatedStringHandler.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#if NET6_0_OR_GREATER
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Build.Framework
+{
+    [InterpolatedStringHandler]
+    public ref struct LogInterpolatedStringHandler
+    {
+        private char[] buffer;
+        private int position = 0;
+        private int argPosition = 0;
+
+        public object[] Arguments { get; } = Array.Empty<object>();
+
+        public LogInterpolatedStringHandler(int literalLength, int formattedCount)
+        {
+            if (formattedCount > 99)
+            {
+                throw new ArgumentOutOfRangeException("Number of formatted arguments must be less than 100.");
+            }
+
+            // Length is computed with reserved space for "{x}" and "{xx}" placeholders 
+            buffer = new char[literalLength + (4 * formattedCount)];
```

Comment:
Is this problematic if you want something like $"my{0}literal{0}string{0}", right?

Reply:
rather this:
```
var longString = "loooooooooooooooooooooooooooooooooooooooooooooooooong";
Log.LogMessage($"msg: {longString}");
```

I believe we cannot avaoid reallocations. How is this handled in the default interpolation handler in runtime?

Reply:
Oh I see now - we are actually translating interpolated string into formatted string - interresting. Than I take it back - all good.

> Is this problematic if you want something like $"my{0}literal{0}string{0}"?
AFAIK each eargument in interpolated string is a separate argument - your example likely translates to `Arguments` having 3 items (all being literal `0`).
Still a fair question and worth explicit unit test.

Reply:
I believe it's not a problem. ```literalLength``` is a length of string when its interpolated. Its designed that way to avoid reallocation. I added more space (4 times parameter count) to reserve characters for {x} and {xx} tokens as I have to create string format for the event.